osc.js

```diff
@@ -1,87 +1,574 @@
-var instance_skel = require('../../instance_skel')
-var GetUpgradeScripts = require('./upgrades')
-var OSC = require('osc')
+const instance_skel = require('../../instance_skel')
+const GetUpgradeScripts = require('./upgrades')
+const OSC = require('osc')
 
-function instance(system, id, config) {
-	var self = this
+class instance extends instance_skel {
+	constructor(system, id, config) {
+		// super-constructor
+		super(system, id, config)
+		this.actions() // export actions
+		return this
+	}
 
-	// super-constructor
-	instance_skel.apply(this, arguments)
+	GetUpgradeScripts = GetUpgradeScripts
 
-	self.actions() // export actions
+	updateConfig(config) {
+		this.config = config
+	}
 
-	return self
-}
+	/**
+	 * This is a method that initializes the instance for companion
+	 * it needds to set status after successful execution
+	 */
+	init() {
+		this.osc = this.osc_server_init()
```

Comment:
Assignment to `this.osc` is not necessary given assignment already occurs in the called method.